### ISSUE TITLE
`x.py setup` fails with explicit panic

### ISSUE THREAD

**Issue by Joshument** [rust-lang/rust#102555]
When trying to run `x.py setup`, the program exits with an explicit panic and does not create the `config.toml` file. The error and full backtrace are as follows (on Arch Linux x86_64):
```
Building rustbuild
    Finished dev [unoptimized] target(s) in 0.03s
Welcome to the Rust project! What do you want to do with x.py?
a) library: Contribute to the standard library
b) compiler: Contribute to the compiler itself
c) codegen: Contribute to the compiler, and also modify LLVM or codegen
d) tools: Contribute to tools which depend on the compiler, but do not modify it directly (e.g. rustdoc, clippy, miri)
e) user: Install Rust from source
Please choose one (a/b/c/d/e): b
thread 'main' panicked at 'explicit panic', builder.rs:821:17
stack backtrace:
   0:     0x564136529f00 - std::backtrace_rs::backtrace::libunwind::trace::he4b135b282ad0bdb
                               at /rustc/2a65764f21cf10b7e03c645f1ad0946e42758e0a/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   1:     0x564136529f00 - std::backtrace_rs::backtrace::trace_unsynchronized::h95dd1a9f59a261f0
                               at /rustc/2a65764f21cf10b7e03c645f1ad0946e42758e0a/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x564136529f00 - std::sys_common::backtrace::_print_fmt::hb2d5f7f2edfb1871
                               at /rustc/2a65764f21cf10b7e03c645f1ad0946e42758e0a/library/std/src/sys_common/backtrace.rs:66:5
   3:     0x564136529f00 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::hbbdb63e1a9e5bae0
                               at /rustc/2a65764f21cf10b7e03c645f1ad0946e42758e0a/library/std/src/sys_common/backtrace.rs:45:22
   4:     0x564136551afe - core::fmt::write::hc0fd3d64c7964deb
                               at /rustc/2a65764f21cf10b7e03c645f1ad0946e42758e0a/library/core/src/fmt/mod.rs:1202:17
   5:     0x564136524ca5 - std::io::Write::write_fmt::h1519d1f8a9f30a76
                               at /rustc/2a65764f21cf10b7e03c645f1ad0946e42758e0a/library/std/src/io/mod.rs:1679:15
   6:     0x56413652c053 - std::sys_common::backtrace::_print::h024f4799bdf90945
                               at /rustc/2a65764f21cf10b7e03c645f1ad0946e42758e0a/library/std/src/sys_common/backtrace.rs:48:5
   7:     0x56413652c053 - std::sys_common::backtrace::print::h7afbde668e15046e
                               at /rustc/2a65764f21cf10b7e03c645f1ad0946e42758e0a/library/std/src/sys_common/backtrace.rs:35:9
   8:     0x56413652c053 - std::panicking::default_hook::{{closure}}::hd91c46b08dd5ed53
                               at /rustc/2a65764f21cf10b7e03c645f1ad0946e42758e0a/library/std/src/panicking.rs:295:22
   9:     0x56413652bd3f - std::panicking::default_hook::hd24a3e5b374ef38d
                               at /rustc/2a65764f21cf10b7e03c645f1ad0946e42758e0a/library/std/src/panicking.rs:314:9
  10:     0x56413652c6fa - std::panicking::rust_panic_with_hook::h227f8f44fe4dcc4d
                               at /rustc/2a65764f21cf10b7e03c645f1ad0946e42758e0a/library/std/src/panicking.rs:698:17
  11:     0x56413652c5b1 - std::panicking::begin_panic_handler::{{closure}}::h643d0c4e5ea667c5
                               at /rustc/2a65764f21cf10b7e03c645f1ad0946e42758e0a/library/std/src/panicking.rs:586:13
  12:     0x56413652a3ac - std::sys_common::backtrace::__rust_end_short_backtrace::hfc98d21e90a8d7a6
                               at /rustc/2a65764f21cf10b7e03c645f1ad0946e42758e0a/library/std/src/sys_common/backtrace.rs:138:18
  13:     0x56413652c312 - rust_begin_unwind
                               at /rustc/2a65764f21cf10b7e03c645f1ad0946e42758e0a/library/std/src/panicking.rs:584:5
  14:     0x564135c3a223 - core::panicking::panic_fmt::h7c86307091386a9a
                               at /rustc/2a65764f21cf10b7e03c645f1ad0946e42758e0a/library/core/src/panicking.rs:142:14
  15:     0x564135c3a0ed - core::panicking::panic::h23b207df3263cf9f
                               at /rustc/2a65764f21cf10b7e03c645f1ad0946e42758e0a/library/core/src/panicking.rs:48:5
  16:     0x56413609dae5 - bootstrap::builder::Builder::new::hec49ddb1e7ccf756
                               at /home/josh/Documents/VsCode/Rust/rust/src/bootstrap/builder.rs:821:17
  17:     0x564135ccf382 - bootstrap::Build::build::h4dff643b0867023d
                               at /home/josh/Documents/VsCode/Rust/rust/src/bootstrap/lib.rs:658:18
  18:     0x564135c3b08c - bootstrap::main::h0575736a7a8a35ba
                               at /home/josh/Documents/VsCode/Rust/rust/src/bootstrap/bin/main.rs:50:5
  19:     0x564135c3c2f3 - core::ops::function::FnOnce::call_once::ha85b876461d50400
                               at /rustc/2a65764f21cf10b7e03c645f1ad0946e42758e0a/library/core/src/ops/function.rs:248:5
  20:     0x564135c3c519 - std::sys_common::backtrace::__rust_begin_short_backtrace::h737bf80906a2863c
                               at /rustc/2a65764f21cf10b7e03c645f1ad0946e42758e0a/library/std/src/sys_common/backtrace.rs:122:18
  21:     0x564135c3cff9 - std::rt::lang_start::{{closure}}::h59767c52524af4f2
                               at /rustc/2a65764f21cf10b7e03c645f1ad0946e42758e0a/library/std/src/rt.rs:166:18
  22:     0x56413651dd7f - core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once::hc94483205bc71284
                               at /rustc/2a65764f21cf10b7e03c645f1ad0946e42758e0a/library/core/src/ops/function.rs:283:13
  23:     0x56413651dd7f - std::panicking::try::do_call::haa04c06f92e34058
                               at /rustc/2a65764f21cf10b7e03c645f1ad0946e42758e0a/library/std/src/panicking.rs:492:40
  24:     0x56413651dd7f - std::panicking::try::h5948f6703ca9775f
                               at /rustc/2a65764f21cf10b7e03c645f1ad0946e42758e0a/library/std/src/panicking.rs:456:19
  25:     0x56413651dd7f - std::panic::catch_unwind::h52dcfc8399de3a68
                               at /rustc/2a65764f21cf10b7e03c645f1ad0946e42758e0a/library/std/src/panic.rs:137:14
  26:     0x56413651dd7f - std::rt::lang_start_internal::{{closure}}::h1fffe4b37fe998da
                               at /rustc/2a65764f21cf10b7e03c645f1ad0946e42758e0a/library/std/src/rt.rs:148:48
  27:     0x56413651dd7f - std::panicking::try::do_call::h771f2fee565f25f2
                               at /rustc/2a65764f21cf10b7e03c645f1ad0946e42758e0a/library/std/src/panicking.rs:492:40
  28:     0x56413651dd7f - std::panicking::try::h7bc1e762d5b9af82
                               at /rustc/2a65764f21cf10b7e03c645f1ad0946e42758e0a/library/std/src/panicking.rs:456:19
  29:     0x56413651dd7f - std::panic::catch_unwind::h6660f137bf3c4a00
                               at /rustc/2a65764f21cf10b7e03c645f1ad0946e42758e0a/library/std/src/panic.rs:137:14
  30:     0x56413651dd7f - std::rt::lang_start_internal::h3b9b5a43f1b6b3ca
                               at /rustc/2a65764f21cf10b7e03c645f1ad0946e42758e0a/library/std/src/rt.rs:148:20
  31:     0x564135c3cfd7 - std::rt::lang_start::hc85f891cccf9dcc8
                               at /rustc/2a65764f21cf10b7e03c645f1ad0946e42758e0a/library/std/src/rt.rs:165:17
  32:     0x564135c3b791 - main
  33:     0x7fd936aee290 - <unknown>
  34:     0x7fd936aee34a - __libc_start_main
  35:     0x564135c3a555 - _start
                               at /build/glibc/src/glibc/csu/../sysdeps/x86_64/start.S:115
  36:                0x0 - <unknown>
Build completed unsuccessfully in 0:00:01
```

**Comment by jyn514** [comment]
This is a regression from https://github.com/rust-lang/rust/pull/101969. We can't create a `Build` until the Setup subcommand has been handled. The fix is to move the `initial_rustfmt` call a little later to line 671.
https://github.com/rust-lang/rust/blob/56a35bc906be1250a76fdb9a4b70e9261e10aec5/src/bootstrap/lib.rs#L658

**Comment by compiler-errors** [comment]
I think this was fixed in #102557?
